rs/rosetta-api/icp/client/src/lib.rs

```diff
@@ -397,6 +398,74 @@ impl RosettaClient {
         }])
     }
 
+    pub fn build_add_hot_key_operations(
+        signer_principal: Principal,
+        neuron_index: u64,
+        public_key: Option<PublicKey>,
+        principal_id: Option<PrincipalId>,
+    ) -> anyhow::Result<Vec<Operation>> {
+        Ok(vec![Operation {
+            operation_identifier: OperationIdentifier {
+                index: 0,
+                network_index: None,
+            },
+            related_operations: None,
+            type_: "ADD_HOTKEY".to_string(),
+            status: None,
+            account: Some(rosetta_core::identifiers::AccountIdentifier::from(
+                AccountIdentifier::new(PrincipalId(signer_principal), None),
+            )),
+            amount: None,
+            coin_change: None,
+            metadata: Some(
+                KeyMetadata {
+                    neuron_index,
+                    key: match (public_key, principal_id) {
+                        (Some(public_key), None) => PublicKeyOrPrincipal::PublicKey(public_key),
+                        (None, Some(principal_id)) => PublicKeyOrPrincipal::Principal(principal_id),
+                        _ => bail!("Either public key or principal id has to be set"),
+                    },
+                }
+                .try_into()
+                .map_err(|e| anyhow::anyhow!("Failed to convert metadata: {:?}", e))?,
+            ),
+        }])
+    }
+
+    pub fn build_remove_hot_key_operations(
```

Comment:
It seems to me this function is exactly the same as `build_add_hot_key_operations` except for the `type_` field. Could we factor this out into a separate generic function?

Reply:
We could, however I designed these build functions specifically for the purposes of readability for external users. The plan is to release the ICP Rosetta client to crates.io and the build operations are essential for the understanding of how transactions with ICP Rosetta are built. extracting logic here adds complexity and reduces readability while shortening the number of lines of codes slightly. I do not think that the benefit outway the cons here.